src/util/make_const_iterator.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <iterator>
+
+template<typename C, typename I>
+typename C::const_iterator make_const_iterator(const C&, const I& it) {
+    return static_cast<typename C::const_iterator>(it);
+}
+
+template<typename C, typename I>
+typename C::iterator make_iterator(C* pContainer, I it) {
+    return std::next(pContainer->begin(), std::distance(pContainer->cbegin(), it));
+}
```

Comment:
This makes an iterator from a const iterator. We need that function as a compatibility layer for the Qt5 API. 
A direct cast is not possible.

Reply:
Answering this question with a PR comment is pointless. Please alter the code to reveal the purpose, e.g. `make_non_const_iterator(...)`.

Btw, renaming this function reveals that the file name is inappropriate and confusing now.

Reply:
Well, the file name still applies if these are supposed to be utility function around *const iterators*. Could be left as is.

Reply:
I see, but wouldn't it not make more sense instead to just change the callsites to obtain a mutable iterator from the start instead of reconstructing it?

Reply:
This is used as a wrapper to make Qt5 functions behave like Qt6 functions ina a way that we can easily drop it in a future version. 

Your suggestion will lead into the other way round which is harder to drop later. 

I consider the perfomance penalty as neglectible as we aim for a Qt6 release in all our targets anyway.

Reply:
I see, I'll need some time to take a closer look at the Qt6 API to properly review this.